cppcache/integration/test/CMakeLists.txt

```diff
@@ -53,11 +55,15 @@ add_executable(cpp-integration-test
   SslTwoWayTest.cpp
   StructTest.cpp
   TransactionCleaningTest.cpp
-  WanDeserializationTest.cpp
-  PdxInstanceFactoryTest.cpp
-  CacheWriterTest.cpp
   TransactionsTest.cpp
+  WanDeserializationTest.cpp
 )
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  target_compile_options(cpp-integration-test
+    PUBLIC
+      -Wno-used-but-marked-unused
```

Comment:
Is this only in a single test? Perhaps we can just add it as a `#pragma` to that file or eve suppress the single line. Worst case we could add this as an option to just that tests cpp file options.

Reply:
It is used twice in two test files, [CommitConflictExceptionTest Line 61](https://github.com/apache/geode-native/blob/6122245ddf0c55fd1e07ae1daae25d67db7889be/cppcache/integration/test/CommitConflictExceptionTest.cpp#L61), [CommitConflictExceptionTest Line 91](https://github.com/apache/geode-native/blob/6122245ddf0c55fd1e07ae1daae25d67db7889be/cppcache/integration/test/CommitConflictExceptionTest.cpp#L91), [ExampleTest Line 102](https://github.com/apache/geode-native/blob/6122245ddf0c55fd1e07ae1daae25d67db7889be/cppcache/integration/test/ExampleTest.cpp#L118), and [ExampleTest Line 102](https://github.com/apache/geode-native/blob/6122245ddf0c55fd1e07ae1daae25d67db7889be/cppcache/integration/test/ExampleTest.cpp#L118).

Since it was used four times across two files I went for cmake to suppress it for just tests.  I'm cool with a pragma if folks feel that's better, I don't have strong feelings.